import.js

```diff
@@ -25,39 +24,38 @@ if (directory.slice(-1) !== '/') {
   directory = directory + '/';
 }
 
+// types must be in highest to lowest level order
+// see https://github.com/whosonfirst/whosonfirst-placetypes
+// venue data goes last
 var types = [
-  'borough',
   'continent',
   'country',
-  'county',
   'dependency',
   'disputed',
+  'macroregion',
+  'region',
+  'county',
+  'macrocounty',
   'localadmin',
   'locality',
-  'macrocounty',
-  'macroregion',
-  'neighbourhood',
-  'region'
+  'borough',
+  'neighbourhood'
 ];
 
-var wofRecords = {};
-
-readStream(directory, types, wofRecords, function() {
-  console.log(Object.keys(wofRecords).length + ' records loaded');
-
-  // a stream of WOF records
-  var recordStream = wofRecordStream.createWofRecordsStream(wofRecords);
+// a cache of only admin records, to be used to fill the hierarchy
+// of other, lower admin records as well as venues
+var wofAdminRecords = {};
 
-  // how to convert WOF records to Pelias Documents
-  var documentGenerator = peliasDocGenerators.create(
-    hierarchyFinder.hierarchies_walker(wofRecords));
+var readStream = readStream.create(directory, types, wofAdminRecords);
```

Comment:
It's a bit confusing that you redefine `readStream` here and assign `readStream.create()` to it. Would be great if this variable or the one in the require block had a different name.